src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +19,58 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession().getSession();
+            transaction = session.beginTransaction();
+            session.save(phone);
+            transaction.commit();
+            return phone;
+        } catch (HibernateException exception) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't create phone " + phone, exception);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder cb = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> query = cb.createQuery(Phone.class);
+            Root<Phone> root = query.from(Phone.class);
+            Predicate conjunctionOfClauses = cb.conjunction();
+            for (Map.Entry<String, String[]> entry : params.entrySet()) {
+                CriteriaBuilder.In<String> inClause = cb.in(root.get(entry.getKey()));
+                for (String option : entry.getValue()) {
+                    inClause.value(option);
+                }
+                conjunctionOfClauses = cb.and(conjunctionOfClauses, inClause);
+            }
+            query.where(conjunctionOfClauses);
+            return session.createQuery(query).getResultList();
+        } catch (HibernateException exception) {
+            throw new RuntimeException("Couldn't find all phones with params: "
+                    + convertToString(params));
+        }
+    }
+
+    private String convertToString(Map<String, String[]> params) {
```

Comment:
let's remove this method, it's redundant. Also, for the printing of this info, you could use stream API